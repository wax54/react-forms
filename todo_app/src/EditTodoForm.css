.EditTodoForm-title{
    margin: 1rem;
}