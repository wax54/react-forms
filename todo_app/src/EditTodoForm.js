import React, { useState } from "react";
import './EditTodoForm.css';

const EditTodoForm = ({ onSubmit, content, id }) => {
    const INITIAL_STATE = content;
    const [title, setTitle] = useState(INITIAL_STATE);
    const handleSubmit = evt => {
        evt.preventDefault();
        onSubmit({id, title});
        setTitle(INITIAL_STATE);
    }

    const handleChange = (evt) => setTitle(evt.target.value);

    return (
        <form className="EditTodoForm" onSubmit={handleSubmit}>
            <label htmlFor="title-input"></label>
            <input
                data-testid="EditTodoForm-input"
                className="EditTodoForm-title"
                id="title-input"
                type="text"
                name="title"
                value={title}
                onChange={handleChange}
                placeholder={content}
                required
            />
            <button>Update</button>

        </form>
    )
};
export default EditTodoForm