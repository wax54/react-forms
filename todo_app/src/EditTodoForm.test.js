import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import EditTodoForm from './EditTodoForm';

it("Doesn't Blow up", () => {
    render(<EditTodoForm />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<EditTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});

it("input should pre populate with task", () => {
    const testTitle = "DEFAULT TASK";

    const { getByTestId } = render(<EditTodoForm onSubmit={() => undefined} content={testTitle} id={1}/>);

    const titleInput = getByTestId("EditTodoForm-input");

    expect(titleInput.value).toEqual(testTitle);
});

it("should allow users to type into the inputs", () => {
    const testTitle = "UPDATED TASK";
    const defaultTitle = "DEFAULT TASK";

    const { getByTestId } = render(<EditTodoForm onSubmit={() => undefined} content={defaultTitle} id={1} />);

    const titleInput = getByTestId("EditTodoForm-input");

    fireEvent.change(titleInput, { target: { value: testTitle } });
    expect(titleInput.value).toEqual(testTitle);
});

it("should pass input data to the passed function when submited", () =>{
    const testTitle = "UPDATED TASK";
    const defaultTitle = "DEFAULT TASK";

    let submittedData;
    const mockedOnSubmit = (data) => submittedData = data;

    const { getByText, getByTestId } = render(<EditTodoForm onSubmit={mockedOnSubmit} content={defaultTitle} id={1} />);

    const titleInput = getByTestId("EditTodoForm-input");
    fireEvent.change(titleInput, { target: { value: testTitle } });

    const btn = getByText("Update");

    fireEvent.click(btn);

    expect(submittedData).toEqual({title: testTitle, id: 1});
});


