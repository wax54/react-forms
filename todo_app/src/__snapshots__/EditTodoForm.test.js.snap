// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <form
    class="EditTodoForm"
  >
    <label
      for="title-input"
    />
    <input
      class="EditTodoForm-title"
      data-testid="EditTodoForm-input"
      id="title-input"
      name="title"
      required=""
      type="text"
      value=""
    />
    <button>
      Update
    </button>
  </form>
</DocumentFragment>
`;
